app/services/message-link-convertor.js

```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+const { getOwner } = Ember;
+
+export default Ember.Service.extend({
+  i18n: Ember.inject.service(),
+
+  convert: function(values) {
+    var offerId = values.offer.id;
+    var msg = values.body;
+    var url_with_text = msg.slice(msg.indexOf("[")+1, msg.indexOf("]"));
+    var url_text_begin = url_with_text.indexOf("|");
+    var url_text = url_with_text.slice(0, url_text_begin);
+    var url_for = url_with_text.slice(url_text_begin+1);
+
+    if(url_for === 'transport_page'){
```

Comment:
I am using if condition for now as we need to check it with 'transport_page' only, later I will use switch or may be key, value pair accordingly.